### PR TITLE
Feature holdem betting

### LINKED ISSUE #47: Req. 2.1.2 - Texas Hold Em Betting Phase
Implement the betting phase of a round of Texas Hold Em

- Each player starting with the player to the left of the big blind is given the option to fold, match the big blind wager, or raise the big blind wager
- This process will continue until all players not folded have placed the same wager

### PR DESCRIPTION
**Pull request by modmo22**
I really just need eyes on this ticket. If someone can help me get the betting logic down, this requirement will be done. Don't worry about AI logic or properly incorporating the pots.

### CODE REVIEW

File: CardsCashCasino/Manager/TexasHoldEmManager.cs [hunk 1]
```diff
@@ -281,6 +360,18 @@ public void Draw(SpriteBatch spriteBatch)
                 hand.Draw(spriteBatch);
             }
         }
+        
+        private Point GetNewCursorPos()
```
Comment (jacobwilki01) [inline CardsCashCasino/Manager/TexasHoldEmManager.cs:374]: summary comment!

File: CardsCashCasino/Manager/TexasHoldEmManager.cs [hunk 2]
```diff
@@ -255,8 +274,68 @@ public void Update()
...
+                        break;
+                }
+
+                _userActionTimeout = new(200);
+                _userActionTimeout.Elapsed += OnTimeoutEvent!;
+                _userActionTimeout.Start();
+            }
         }
 
         private void UpdateWhileAIPlaying()
```
Comment (jacobwilki01) [inline CardsCashCasino/Manager/TexasHoldEmManager.cs:351]: summary comment!

File: CardsCashCasino/Manager/TexasHoldEmManager.cs [hunk 3]
```diff
@@ -266,11 +345,11 @@ private void UpdateWhileAIPlaying()
         public void Draw(SpriteBatch spriteBatch)
```
Comment (jacobwilki01) [inline CardsCashCasino/Manager/TexasHoldEmManager.cs]: add space and summary comment!

File: CardsCashCasino/Manager/TexasHoldEmManager.cs [hunk 4]
```diff
@@ -355,13 +444,161 @@ private void EndGame()
...
         {
-            _playerHands[0] = _userHand;
+            _playerHands.Add(_userHand);
             for (int i = 0; i < Constants.AI_PLAYER_COUNT; i++)
             {
                 _playerHands.Add(new PokerAIHand());
             }
         }
 
+        private void HandleBettingPhase()
```
Comment (jacobwilki01) [inline CardsCashCasino/Manager/TexasHoldEmManager.cs:595]: summary comment

File: CardsCashCasino/Manager/TexasHoldEmManager.cs [hunk 5]
```diff
@@ -355,13 +444,161 @@ private void EndGame()
...
+                case BettingPhase.TURN:
+                    // HandleTurn();
+                    break;
+                case BettingPhase.RIVER:
+                    // HandleRiver();
+                    break;
+            }
+        }
+
+        private void HandlePreflop()
```
Comment (jacobwilki01) [inline CardsCashCasino/Manager/TexasHoldEmManager.cs:614]: summary comment

File: CardsCashCasino/Manager/TexasHoldEmManager.cs [hunk 6]
```diff
@@ -355,13 +444,161 @@ private void EndGame()
...
+                
+                // Handle the player's action.
+                HandlePlayerAction(playerIndex);
+                
+                // Increment the player index.
+                playerIndex = (playerIndex + 1) % _playerHands.Count;
+            }
+        }
+
+        private void HandlePlayerAction(int playerIndex)
```
Comment (jacobwilki01) [inline CardsCashCasino/Manager/TexasHoldEmManager.cs:641]: summary comment

File: CardsCashCasino/Manager/TexasHoldEmManager.cs [hunk 7]
```diff
@@ -355,13 +444,161 @@ private void EndGame()
...
+                case PokerAction.RAISE:
+                    // HandleRaise(playerIndex);
+                    break;
+                case PokerAction.ALL_IN:
+                    // HandleAllIn(playerIndex);
+                    break;
+            }
+        }
+        
+        private PokerAction GetUserAction()
```
Comment (jacobwilki01) [inline CardsCashCasino/Manager/TexasHoldEmManager.cs:671]: summary comment

File: CardsCashCasino/Manager/TexasHoldEmManager.cs [hunk 8]
```diff
@@ -355,13 +444,161 @@ private void EndGame()
...
+                Constants.CHECK_BUTTON_POS => PokerAction.CHECK,
+                Constants.CALL_BUTTON_POS => PokerAction.CALL,
+                Constants.RAISE_BUTTON_POS => PokerAction.RAISE,
+                Constants.FOLD_BUTTON_POS => PokerAction.FOLD,
+                Constants.ALL_IN_BUTTON_POS => PokerAction.ALL_IN,
+                _ => PokerAction.CHECK
+            };
+        }
+        
+        private PokerAction GetAIAction(int playerIndex)
```
Comment (jacobwilki01) [inline CardsCashCasino/Manager/TexasHoldEmManager.cs:684]: summary comment

File: CardsCashCasino/Manager/TexasHoldEmManager.cs [hunk 9]
```diff
@@ -468,4 +705,6 @@ public Point GetAdjustedPos()
         }
         #endregion Methods
     }
+
```
Comment (jacobwilki01) [inline CardsCashCasino/Manager/TexasHoldEmManager.cs:851]: remove whitespace

**Review by jacobwilki01 — APPROVED** [review]

**Review by eBerkley — APPROVED** [review]
As long as it's commented its good

**Pull request merged**